system-addon/content-src/components/Card/Card.jsx

```diff
@@ -131,11 +194,24 @@ export class _Card extends React.PureComponent {
     this.maybeLoadImage();
   }
 
+  componentWillMount() {
+    this.updateCardImage(this.props);
+  }
+
   componentWillReceiveProps(nextProps) {
     // Clear the image state if changing images
-    if (nextProps.link.image !== this.props.link.image) {
+    if (nextProps.link && !this.isImageInState(nextProps.link.image)) {
       this.setState({imageLoaded: false});
     }
+
+    // NB: updateCardImage needs to be called after isImageInState since cardImage's
+    // state is modified in updateCardImage and isImageInState uses the old state.
+    this.updateCardImage(nextProps);
```

Comment:
Separately, @k88hudson I noticed React renamed this method to UNSAFE_componentWillReceiveProps https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops Should we be doing something differently now?